atuin-server/src/database.rs

```diff
@@ -336,6 +337,26 @@ impl Database for Postgres {
         Ok(())
     }
 
+    #[instrument(skip_all)]
+    async fn delete_user(&self, u: &User) -> Result<()> {
+        sqlx::query_as::<_, Session>("delete from sessions where user_id = $1")
+            .bind(u.id)
+            .fetch_all(&self.pool)
+            .await?;
+
+        sqlx::query_as::<_, Session>("delete from users where id = $1")
+            .bind(u.id)
+            .fetch_all(&self.pool)
+            .await?;
+
+        sqlx::query_as::<_, Session>("delete from history where user_id = $1")
+            .bind(u.id)
+            .fetch_all(&self.pool)
+            .await?;
+
+        Ok(())
+    }
```

Comment:
Seeing as we don't need to fetch any data, we can actually just do this instead

```suggestion
    async fn delete_user(&self, u: &User) -> Result<()> {
        sqlx::query::("delete from sessions where user_id = $1")
            .bind(u.id)
            .execute(&self.pool)
            .await?;

        sqlx::query::("delete from users where id = $1")
            .bind(u.id)
            .execute(&self.pool)
            .await?;

        sqlx::query("delete from history where user_id = $1")
            .bind(u.id)
            .execute(&self.pool)
            .await?;

        Ok(())
    }
```